### PR TITLE
Fix the Go toolchain setup for MacOS

### PR DESCRIPTION
**Pull request by erwinvaneyk**
This PR fixes the development setup by ensuring that the correct Go toolchain is downloaded based on OS.

Beyond that it cleans up some unused variables and recipes.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,9 +40,7 @@ SPRINGBOARD_FULL_TAG := $(SPRINGBOARD_REPO_TAG):$(IMAGE_TAG)
 
 DEFAULT_HTTP_IMAGE_TAG ?= platform9systems/decco-default-http
 
-export GOPATH:=$(GOPATH_DIR)
```
Comment (bcle) [inline Makefile:39]: Now that this line is removed, what GOPATH does the build use?

Comment (erwinvaneyk) [reply Makefile:39]: I was thinking it would use the one already set in the system, since the packages are not version or OS-dependent.

Comment (erwinvaneyk) [reply Makefile:39]: But I will read it in case GOPATH is not set at all.

Comment (bcle) [reply Makefile:39]: Ok, I'll have to make sure the TC builders actually have a sane GOPATH defined, or explicitly define one (perhaps a temporary one) in the TC build step. I'm slightly nervous about multiple builds on the same builder sharing the same GOPATH.

Comment (erwinvaneyk) [reply Makefile:39]: Go modules supports multiple versions of a dependency to be stored in the same GOPATH. So it should be fine, as long as some GOPATH is provided.

**Pull request merged**